infer/src/pulse/PulseFormula.ml

```diff
@@ -3905,24 +4008,24 @@ module Formula = struct
         | None ->
             Sat (phi, new_eqs)
         | Some atoms ->
-            and_normalized_atoms (phi, new_eqs) atoms >>| snd
+            and_normalized_atoms (phi, new_eqs) atoms ~orig_atom:atoms ~add_term:true >>| snd
```

Comment:
is this needed? If not then `Normalizer` wouldn't need to care about adding raw atoms and terms and we could leave that task to client code below.